evm_trace/geth.py

```diff
@@ -0,0 +1,232 @@
+import math
+from typing import Dict, Iterator, List
+
+from eth_utils import to_int
+from ethpm_types import BaseModel, HexBytes
+from pydantic import Field, validator
+
+from evm_trace.base import CallTreeNode
+from evm_trace.enums import CallType
+
+
+class TraceFrame(BaseModel):
+    """
+    A low-level data structure modeling a transaction trace frame
+    from the Geth RPC ``debug_traceTransaction``.
+    """
+
+    pc: int
+    """Program counter."""
+
+    op: str
+    """Opcode."""
+
+    gas: int
+    """Remaining gas."""
+
+    gas_cost: int = Field(alias="gasCost")
+    """The cost to execute this opcode."""
+
+    depth: int
+    """
+    The number of external jumps away the initially called contract (starts at 0).
+    """
+
+    stack: List[HexBytes] = []
+    """Execution stack."""
+
+    memory: List[HexBytes] = []
+    """Execution memory."""
+
+    storage: Dict[HexBytes, HexBytes] = {}
+    """Contract storage."""
+
+    @validator("pc", "gas", "gas_cost", "depth", pre=True)
+    def validate_ints(cls, value):
+        return int(value, 16) if isinstance(value, str) else value
+
+
+def get_calltree_from_geth_call_trace(data: Dict) -> CallTreeNode:
+    """
+    Creates a CallTreeNode from a given transaction call trace.
+
+    Args:
+        data (Dict): The response from ``debug_traceTransaction`` when using
+          ``tracer=callTracer``.
+
+    Returns:
+        :class:`~evm_trace.base.CallTreeNode`: Call tree of transaction trace.
+    """
+
+    data = _validate_data_from_call_tracer(data)
+    root = CallTreeNode(**data)
+
+    def fix_depth(r: CallTreeNode):
+        for c in r.calls:
+            c.depth = r.depth + 1
+            fix_depth(c)
+
+    fix_depth(root)
+    return root
+
+
+def get_calltree_from_geth_trace(
```

Comment:
I think someday we should consider renaming this so it is more obvious it is coming from the `structLog` tracer.